javalib/src/main/scala/java/util/concurrent/CopyOnWriteArrayList.scala

```diff
@@ -0,0 +1,337 @@
+package java.util
+package concurrent
+
+import java.lang.{reflect => jlr}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+import scala.scalajs._
+
+class CopyOnWriteArrayList[E] private(private var inner: js.Array[E])
+    extends List[E] with RandomAccess with Cloneable with Serializable {
+  self =>
+
+  // requiresCopyOnWrite is false if and only if no other object
+  // (like the iterator) may have a reference to inner
+  private var requiresCopyOnWrite = false
+
+  def this() = {
+    this(new js.Array[E])
```

Comment:
Ah. Hum the `CopyOnWriteArrayListView` extending `CopyOnWriteArrayList` is suspicious, especially since the `inner` is set to `null` :-s
How much code duplication does this really avoid?